app/controllers/local_transaction_controller.rb

```diff
@@ -0,0 +1,129 @@
+class LocalTransactionController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  before_filter :validate_slug_param, only: :search
```

Comment:
Validating the slug is now a bit weird, because we already use the slug in the routes.rb file with the advanced constraints. So doing it here is happening too late. We've written up a story to push this into middle ware of some sort -> https://trello.com/c/leLnBdG4/580-refactor-frontend-cleanup-slug-validation. So unless this breaks any tests, I'd say remove the line for now.